doc/sphinx/user-extensions/syntax-extensions.rst

```diff
@@ -7,64 +7,179 @@ In this chapter, we introduce advanced commands to modify the way Coq
 
-   .. todo indentation of this chapter is not consistent with other chapters.  Do we have a standard?
+   .. coqtop:: in
 
-For example, the following definition permits using the infix expression :g:`A /\ B`
-to represent :g:`(and A B)`:
+      Notation Nlist := (list nat).
 
-.. coqtop:: in
+   .. coqtop:: all
 
-   Notation "A /\ B" := (and A B).
+      Check 1 :: 2 :: 3 :: nil.
 
-:g:`"A /\ B"` is a *notation*, which tells how to represent the abbreviated term
-:g:`(and A B)`.
+   .. coqtop:: in
+
+      Notation reflexive R := (forall x, R x x).
+
+   .. coqtop:: all
+
+      Check forall A:Prop, A <-> A.
+      Check reflexive iff.
+
+   .. coqtop:: in
+
+      Notation Plus1 B := (Nat.add B 1).
+
+   .. coqtop:: all
+
+      Compute (Plus1 3).
+
+   An abbreviation expects no precedence nor associativity, since it
+   is parsed as a usual application. Abbreviations are used as
+   much as possible by the Coq printers unless the modifier ``(only
+   parsing)`` is given.
+
+   An abbreviation is bound to an absolute name as an ordinary definition is
+   and it also can be referred to by a qualified name.
+
+   Abbreviations are syntactic in the sense that they are bound to
+   expressions which are not typed at the time of the definition of the
+   abbreviation but at the time they are used. Especially, abbreviations
+   can be bound to terms with holes (i.e. with “``_``”). For example:
+
+   .. coqtop:: none reset
+
+      Set Strict Implicit.
+      Set Printing Depth 50.
+
+   .. coqtop:: in
+
+      Definition explicit_id (A:Set) (a:A) := a.
+
+   .. coqtop:: in
+
+      Notation id := (explicit_id _).
+
+   .. coqtop:: all
+
+      Check (id 0).
+
+   Abbreviations disappear when a section is closed. No typing of the
+   denoted expression is performed at definition time. Type checking is
+   done only at the time of use of the abbreviation.
+
+   Like for notations, if the right-hand side of an abbreviation is a
+   partially applied constant, the abbreviation inherits the implicit
+   arguments and notation scopes of the constant. As an
+   exception, if the right-hand side is just of the form :n:`@@qualid`,
+   this conventionally stops the inheritance of implicit arguments.
+
+   Like for notations, it is possible to bind binders in
+   abbreviations. Here is an example:
+
+   .. coqtop:: in reset
+
+      Definition force2 q (P:nat*nat -> Prop) :=
+        (forall n', n' >= fst q -> forall p', p' >= snd q -> P q).
+
+      Notation F p P := (force2 p (fun p => P)).
+      Check exists x y, F (x,y) (x >= 1 /\ y >= 2).
+
+.. _ReservingNotations:
+
+Reserving notations
+-------------------
+
+.. cmd:: Reserved Notation @string {? ( {+, @syntax_modifier } ) }
+
+   Modifies the lexer (by adding keywords) and the parser.
 
 Notations must be in double quotes, except when the
 abbreviation has the form of an ordinary applicative expression;
 see :ref:`Abbreviations`. The notation consists of *tokens* separated by
```

Comment:
I think it's better to keep with the PR's tendency to make abbreviations and notations different beasts.
```suggestion
Notations must be in double quotes, unlike :ref:`Abbreviations`.
The notation consists of *tokens* separated by
```